DOC : Registration CLI (#2272)
dipy/dipy: doc/interfaces/registration_flow.rst

-------------------
Affine Registration
-------------------

Affine registration is an optimization strategy used to maximize the mutual 
information between two volumes or images. The affine transformation allows 
linear coordinate changes that include shearing, scaling, rotation and translation.

The workflow for the affine registration requires the paths to the static image file,
and to the moving image file. Static image is used as a reference and moving image is
transformed during optimization.

You may want to create an output directory to save the transformed image::

    mkdir affine_reg_output

Run the following command::

    dipy_align_affine <path_to_static_file> <path_to_moving_file> --out_dir "affine_reg_output" --out_affine "affine_reg.txt"

This command will apply affine transformation on the moving image file and save the
transformed image and the affine matrix to the directory ``affine_reg_output``.

In case, you did not specify the ``out_dir``, the transformed image file and 
affine matrix would be saved in the input directory by deafult. If you did not
specify  ``out_affine``, the affine matrix will be saved to a file named 

> ``affine.txt`` by default, located in the input directory also by default.

Reviewer: I am not sure about this, but shouldn't this be:

```suggestion
``affine.txt`` by default, located in the output directory also by default.
```

Reviewer 2: Having read the method, I'd say that @arokem is right.

------------------------------------
Symmetric Diffeomorphic Registration
------------------------------------

Symmetric Diffeomorphic Registration is performed using the ``Symmetric Normalization

> (SyN)`` algorithm. It is an optimization technique that brings moving image closer to

Reviewer: ```suggestion
(SyN)`` algorithm. It is an optimization technique that brings the moving image closer to
```

Author: Addressed in  81833cc

the static image.

> Create a directory where to save the transformed image (e.g.: ``syn_reg_output``)::

Reviewer: ```suggestion
Create a directory in which to save the transformed image (e.g.: ``syn_reg_output``)::
```

Author: Reworded in 81833cc

    
    mkdir syn_reg_output

To run the symmetric normalization registration method, we need to specify the 

> paths to the static image file, and to the moving image file, followed by the

Reviewer: ```suggestion
paths to the static image file, and to the moving image file, followed by
```

Author: Addressed in 81833cc

optional arguments. In this case, we will be specifying the ``metric``,
``out_dir`` and ``out_warped``. You can use cc (Cross Correlation), ssd (Sum

> Squared Difference) or em (Expectation-Maximization) as metric.

Reviewer: ```suggestion
Squared Difference) or em (Expectation-Maximization) as metrics.
```

Author: Addressed in 81833cc

> So, we will run the command as::

Reviewer: ```suggestion
We will run the command as::
```

Author: Addressed in db5a59a

    dipy_align_syn <path_to_static_file> <path_to_moving_file> --metric "cc" --out_dir "syn_reg_output" --out_warped "syn_reg_warped.nii.gz"

This command will perform symmetric diffeomorphic registration and save it to 
the specified output directory.

> In case, you did not specify the ``out_dir``, the transformed files would be 

Reviewer: ```suggestion
In case you did not specify the ``out_dir``, the transformed files would be 
```

Author: Addressed in db5a59a

saved in the input directory by deafult. If you did not specify ``out_warped``, 
the warped file will be saved as ``warped_moved.nii.gz`` by default.

----------------------
Apply Registration
----------------------

In order to apply transform registration, we need to specify the path of the 

> static image file, moving image file and transform map file followed by the 

Reviewer: ```suggestion
static image file, moving image file, and transform map file followed by the 
```

Author: Addressed in db5a59a

optional arguments. In this case, we will be specifying the ``transform_type``
and ``out_dir``.

> Create a directory where to save the transfomed files (e.g.: ``transform_reg_output``)::

Reviewer: ```suggestion
Create a directory in which to save the transformed files (e.g.: ``transform_reg_output``)::
```

Author: Addressed in  81833cc, db5a59ac

    mkdir transform_reg_output

Then, run the command as::

    dipy_apply_transform <path_to_static_file> <path_to_moving_file> <path_to_transform_map_file> --transform_type "diffeomorphic" --out_dir "transform_reg_output"

This command will transform the moving image and save the transformed files 
to the specified output directory.

-----------------------------
Streamline-based Registration
-----------------------------

Streamline-based registration is performed to align bundles of streamlines 

> directly inthe space of streamlines. The aim is to align the moving image 

Reviewer: ```suggestion
directly in the space of streamlines. The aim is to align the moving image 
```

Author: Addressed in 81833cc

> directly inthe space of streamlines. The aim is to align the moving image 

Reviewer: ```suggestion
directly inthe space of streamlines. The aim is to align the moving streamlines 
```

Author: Addressed in 81833cc

> with the static image.

Reviewer: ```suggestion
with the static streamlines.
```

Author: Reworded in 81833cc

The workflow for streamline-based registration requires the paths to the 

> static image file, and to the moving image file, followed by the optional 

Reviewer: ```suggestion
static streamlines file, and to the moving streamlines file, followed by the optional 
```

Author: Reworded in 81833cc
